src/main/java/com/flightstats/hub/channel/ChannelResource.java

```diff
@@ -73,13 +73,14 @@
     public Response getChannelMetadata(@PathParam("channel") String channelName,
                                        @QueryParam("cached") @DefaultValue("true") boolean cached) {
         logger.debug("get channel {}", channelName);
-        ChannelConfig config = channelService.getChannelConfig(channelName, cached);
-        if (config == null) {
+
+        Optional<ChannelConfig> optionalChannelConfig = channelService.getChannelConfig(channelName, cached);
+        if (!optionalChannelConfig.isPresent()) {
```

Comment:
If you wanted to get all fancy and maybe less readable, you could do something like
```
if (!optionalChannelConfig.isPresent()) {
    logger.info("unable to get channel " + channelName);
}

return optionalChannelConfig
    .map(config -> buildChannelConfigResponse(config, uriInfo, channelName))
    .map(Response::ok)
    .map(ResponseBuilder(orwhateverthisis?)::build)
    .orElseThrow(...);
```
The logging kinda messes up the flow a little bit, but also that's a lot of maps. I honestly hate calling `Optional.get` though.

Reply:
Oh wait, you might be able to do `orElseThrow(() -> { logger.info(...); return new WebApplicationException(...);)` Up to you. That's also kinda weird.